votor/src/consensus_pool/vote_pool.rs

```diff
@@ -7,51 +7,37 @@ use {
     std::collections::{HashMap, HashSet},
 };
 
-#[derive(Debug)]
-pub(crate) struct VoteEntry {
-    pub(crate) vote_messages: Vec<VoteMessage>,
-    pub(crate) total_stake_by_key: Stake,
-}
-
-impl VoteEntry {
-    pub fn new() -> Self {
-        Self {
-            vote_messages: Vec::new(),
-            total_stake_by_key: 0,
-        }
-    }
-}
-
-pub(crate) trait VotePool {
-    fn total_stake(&self) -> Stake;
-    fn has_prev_validator_vote(&self, validator_vote_key: &Pubkey) -> bool;
+#[derive(Default)]
+struct VoteEntry {
+    vote_messages: Vec<VoteMessage>,
+    total_stake_by_key: Stake,
 }
 
 /// There are two types of vote pools:
 /// - SimpleVotePool: Tracks all votes of a specfic vote type made by validators for some slot N, but only one vote per block.
 /// - DuplicateBlockVotePool: Tracks all votes of a specfic vote type made by validators for some slot N,
 ///   but allows votes for different blocks by the same validator. Only relevant for VotePool's that are of type
 ///   Notarization or NotarizationFallback
-pub(crate) enum VotePoolType {
+pub(super) enum VotePool {
```

Comment:
I don't have strong feelings about this, and I agree that `vote_pools: BTreeMap<PoolId, VotePool>,` sounds more natural, but I think we originally named it `VotePoolType` because `VotePoolType::SimpleVotePool` sounds a bit more natural.